app/controllers/provider/admin/messages/inbox_controller.rb

```diff
@@ -22,11 +22,12 @@ def reply
     reply = @message.reply
     reply.attributes = message_params
 
-    reply.save!
-    reply.deliver!
-
-    flash[:notice] = 'Reply was sent.'
-    redirect_to action: :index
+    if reply.valid?
+      send_reply(reply)
+    else
+      flash[:error] = reply.errors.full_messages.to_sentence
+      redirect_to provider_admin_messages_inbox_path(@message)
+    end
```

Comment:
We could remote the `send_reply` method by rewriting this like
```ruby
def reply
  reply = @message.reply
  reply.attributes = message_params

  if reply.save && reply.deliver
    flash[:notice] = 'Reply was sent.'
    redirect_to action: :index
  else
    flash[:error] = reply.errors.full_messages.to_sentence
    redirect_to provider_admin_messages_inbox_path(@message)
  end
end
```
"Advantage" is to remove one method.

Reply:
this will fails due to code climate

Reply:
> this will fails due to code climate

Do you know what issue CodeClimate has with this? I don't think it hurts Rubocop nor Reek 🤔 

 > we must check if message is valid then only we will save and get it delivered

We can check it with `valid?` as we're doing of course. I just want to point out that by using `if reply.save && reply.deliver`, if `reply.save` fails (due to validation), it will return `false` and `reply.deliver` won't even be executed.

Reply:
Okay agreed and updated, let me just check if any rubocop or reek fails